First pass for the stash origin documentation (SOFTWARE-3900) (#681)
opensciencegrid/docs: docs/data/stashcache/run-stash-origin-container.md

provides organizations with a method to distribute their data in a scalable manner to thousands of jobs without needing
to pre-stage data across sites or operate their own scalable infrastructure.

[Stash Origins](/data/stashcache/install-origin) store copies of users' data. Each community (or experiment) needs to run one origin to export its data via the StashCache federation.
This document outlines how to run such an origin in a Docker container.

Before Starting
---------------

Before starting the installation process, consider the following points:

1. **Docker:** For the purpose of this guide, the host must have a running docker service and you must have the ability to start containers (i.e., belong to the `docker` Unix group).
1. **Network ports:** The Stash Origin listens for incoming HTTP/S and XRootD connections on ports 1094 and 1095 (by default).
1. **File Systems:** Stash Origin needs a partition on the host to store user data.

Configuring Stash Origin
------------------------

In addition to the required configuration above (ports and file systems), you may also configure the behavior of your origin with the following variables using an environment variable file:

> Where the environment file on the docker host, `/opt/origin/.env`, has (at least) the following contents (replace "YOUR_SITE_NAME" with the name of your site as [registered in Topology](/stashcache/install-origin/#registering-the-origin):

Reviewer: ```suggestion
Where the environment file on the docker host, `/opt/origin/.env`, has (at least) the following contents (replace "YOUR_SITE_NAME" with the name of your site as [registered in Topology](/data/stashcache/install-origin/#registering-the-origin):
```

1. **File Systems:** Stash Cache needs a partition on the host to store cached data

Configuring Stashcache
----------------------

In addition to the required configuration above (ports and file systems), you may also configure the behavior of your cache with the following variables using an enviroment variable file:

Where the environment file on the docker host, `/opt/origin/.env`, has (at least)the following contents:
```file
XC_ORIGINEXPORT=/origin
XC_RESOURCENAME = YOUR_SITE_NAME
```

### Creating an Auth file ###

XrootD needs an authentication file [(AuthFile)](/data/xrootd/xrootd-authorization/) to control access to different parts of the namespace in your origin. You can create a simple authfile named `/opt/origin/auth_file` as follows:

```file
u * /origin rl
```

Create a configuration file for XrootD to find your authfile. Create a file `/opt/origin/10-origin-authfile.cfg`:

```
set StashOriginPublicAuthfile = /etc/xrootd/public-origin-authfile
```

### Disabling OSG monitoring ###

By default, XCache reports to the OSG so that OSG staff can monitor the health of data federations.

> If you would like to report monitoring information to another destination, you can disable the OSG monitoring by setting

Reviewer: Do you _have to_ disable OSG reporting to report somewhere else, or can you report to two places?

Author: Yes you can but that is kinda advanced. So it should be either monitoring or nothing else

Reviewer: So how about something like
```suggestion
To disable OSG monitoring (for example so you can report monitoring information to another destination), set
```
